### DISCUSSION TITLE
3.0 已发布正式版

### DISCUSSION THREAD

**Discussion by tangjinzhou** [vueComponent/ant-design-vue#4575] [Show and tell]
#### About v2

We have already supported vue3, but it is not a completely refactored project. There is a lot of room for improvement in performance and experience. Based on this, we started the refactoring of the V3 version, which will have better performance and user experience, and of course it will also bring some breaking change.

# v3.0

## Overview:
   
refactor by typescript & vue 3 composition api，follow the latest design specifications of antd 4.x

## Objectives:

Improve framework performance

Enhance Typescript support

Update the Documentation with improved examples and more information

## Progress

- [x] Affix
- [x] Anchor
- [x] AutoComplete
- [x] Alert
- [x] Avatar
- [x] BackTop
- [x] Badge
- [x] Breadcrumb
- [x] Button
- [x] Col
- [x] Divider
- [x] Icon
- [x] Layout
- [x] Menu
- [x] Space
- [x] typography
- [x] Row
- [x] Select
- [x] Spin
- [x] Skeleton
- [x] Comment
- [x] ConfigProvider
- [x] Empty
- [x] Result
- [x] Descriptions
- [x] PageHeader
- [x] Calendar
- [x] DatePicker
- [x] Form
- [x] Transfer
- [x] Switch
- [x] Tooltip
- [x] Mentions
- [x] Progress
- [x] Rate
- [x] Statistic
- [x] List
- [x] Tag
- [x] TimePicker
- [x] Tree
- [x] TreeSelect
- [x] Slider
- [x] Pagination
- [x] Timeline
- [x] Steps
- [x] Collapse
- [x] Popconfirm
- [x] Popover
- [x] Table
- [x] Input
- [x] Card
- [x] Checkbox
- [x] InputNumber
- [x] Radio
- [x] Dropdown
- [x] Tabs
- [x] Drawer
- [x] message
- [x] notification
- [x] Modal
- [x] Cascader
- [x] Upload
- [x] Carousel

**Comment by wensonsmith** [comment]
表格开箱支持虚拟滚动，希望这个可以尽早发布 。 谢谢！

@surely-vue/table 功能能够直接集成进来就更好了

**Reply by yanbowe** [reply]
souldly-vue/table git地址怎么找不到了

**Reply by maicWorkGithub** [reply]
surely-vue/table 收费了

**Comment by tangjinzhou** [comment]
V3 will use dayjs instead of moment

**Comment by wsxe9988** [comment]
V3 的table是啥样的呢，会不会是surely-vue/table

**Reply by tangjinzhou** [reply]
不会   长的会和 antd react 差不多

**Reply by wsxe9988** [reply]
好吧。不过surely-vue/table 这个确实很不错

**Reply by yanbowe** [reply]
> 好吧。不过surely-vue/table 这个确实很不错

surely-vue/table有文档地址贴一下吗，git找不到了

**Reply by maicWorkGithub** [reply]
surely-vue/table 收费了

**Comment by wsxe9988** [comment]
ColorPicker will exists?

**Reply by tangjinzhou** [reply]
maybe 3.1

**Comment by wsxe9988** [comment]
3.0版的Modal  能否加上以下feature：1.增加上下左右等8个方法的大小伸缩 2.可自由拖动 3.关闭按钮旁边增加全屏放大按钮 4.也可以双击Modal的标题栏切换全屏和退出全屏

**Reply by renqian805** [reply]
这里有个[repo](https://github.com/buqiyuan/vue3-antd-admin)对Modal做了这些扩展，看了实现还挺简单的，可以参考一下

**Comment by wsxe9988** [comment]
![image](https://user-images.githubusercontent.com/5828057/132156095-74d6f5dc-0e9d-4072-91ff-b90562b2f040.png)
3.0的datepicker  能不能实现多个日期的选择呢

**Comment by Hccake** [comment]
v2 到 v3 的破坏性改动多吗？或者说如果项目使用 v2 开发，后续升级 v3 改动大吗

**Reply by tangjinzhou** [reply]
> v2 到 v3 的破坏性改动多吗？或者说如果项目使用 v2 开发，后续升级 v3 改动大吗

尽可能都兼容了

**Comment by wsxe9988** [comment]
v\
![image](https://user-images.githubusercontent.com/5828057/133894058-bdfeec12-4fe9-4352-824c-7fe68363a874.png)
3.0能否考虑增加个这个组件  Screening 筛选控件   可以参考 https://open-hand.github.io/choerodon-ui/zh/procmp/data-entry/screening

**Reply by tangjinzhou** [reply]
table 自带筛选

**Comment by wsxe9988** [comment]
![image](https://user-images.githubusercontent.com/5828057/133894104-ac66db75-d59d-4d34-8212-d53f4f9f0911.png)
Lov 弹出选择输入框   https://open-hand.github.io/choerodon-ui/zh/procmp/data-entry/lov

**Reply by tangjinzhou** [reply]
select

**Reply by Hccake** [reply]
这个自己封装下就好了，可以参考：

代码：https://github.com/ballcat-projects/ballcat-ui-vue/blob/master/src/components/Lov/LovLocal.vue
doc：http://www.ballcat.cn/guide/front/lov-local.html#%E7%AE%80%E4%BB%8B

**Comment by wsxe9988** [comment]
![image](https://user-images.githubusercontent.com/5828057/133894127-b9485ff9-fbf1-41b5-890e-9f2708d97cbf.png)
https://open-hand.github.io/choerodon-ui/zh/procmp/data-entry/attachment

**Reply by tangjinzhou** [reply]
upload

**Comment by wsxe9988** [comment]
![1632151839(1)](https://user-images.githubusercontent.com/5828057/134029645-4fb33f7d-0ca8-4e8d-b6b2-3f244241a532.png)
3.0的table组件能不能对相关字段做补充说明  列增加个help的属性配置

**Comment by yehoan** [comment]
treeSelect 能否新增全反选得功能呢 当数据过多得时候 操作起来很不方便

**Comment by ylc395** [comment]
2.x 还继续发小版本么，还是不再迭代，直接来到 3.x

**Reply by tangjinzhou** [reply]
短期会发阻塞性质的 bugfix   3.x 相对稳定后，会归档 2.x

**Comment by noecs** [comment]
要是能像1.7版本一样使用动态的icon图表就好了。
`<a-icon type="动态动态动态" />`

**Reply by tangjinzhou** [reply]
一直都有

**Comment by wsxe9988** [comment]
高性能表格项目下线了吗 https://github.com/surely-vue/table 找不到了   https://store.antdv.com/pro/preview/list/big-table-list  404了

**Reply by maicWorkGithub** [reply]
surely-vue/table 收费了

**Comment by fleximus** [comment]
[Nuxt 3](https://v3.nuxtjs.org/) is currently in beta. Please support it, ideally as module (see current [Nuxt 3 Beta Module Support](https://github.com/nuxt/framework/discussions/751)).

**Comment by CamouflageJ** [comment]
我想问一下，<a-cascader> - did you register the component correctly?  级联选择是不兼容了嘛？
![image](https://user-images.githubusercontent.com/64939547/144532671-fa34d4d1-5cc4-48e7-be2c-2ff9e9129c02.png)

**Reply by sendya** [reply]
你没有注册组件

**Comment by now915** [comment]
请问下 3.x release 大概什么时候能发布

**Reply by tangjinzhou** [reply]
2022 Q1

**Reply by wsxe9988** [reply]
最近感觉antv 比较安静。还有没重构完成的组件大概什么时候完成呢

**Comment by wsxe9988** [comment]
嘿嘿，有没有考虑像ant-design-mobile似的，出个ant-design-vue-mobile

**Reply by tangjinzhou** [reply]
没有

**Comment by kimmy-wang** [comment]
2.0还维护吗? menu组件在ssr下不显示的问题解决了？

**Reply by tangjinzhou** [reply]
update to 3.0

**Reply by kimmy-wang** [reply]
2.x中notification的description的类型是VNodeTypes ，而3.x notification的description的类型是VNode或者是() => VNode，文档没有更新？

**Reply by tangjinzhou** [reply]
welcome pr

**Comment by Ludidi** [comment]
期望Cascader组件支持多选

**Comment by ILfengzi** [comment]
上传希望实现拖拽文件夹，参考：https://github.com/lian-yue/vue-upload-component

**Comment by cheaprYang** [comment]
css var 3.0版本 什么时候会支持呢？

**Reply by tangjinzhou** [reply]
应该快了

**Reply by kimmy-wang** [reply]
年前能发正式版吗？

**Reply by shao-kang** [reply]
这个功能开发进度怎么样了， 如果没有人员安排，我可以帮忙开发这个功能

**Comment by shao-kang** [comment]
有支持 css 变量自定义主题的计划吗？ 如果有计划，但暂时没有人员安排，我可以来开发提交这个特性

**Reply by zkwolf** [reply]
我来吧，搞得差不多了

**Reply by cheaprYang** [reply]
加油 这样换肤 就方便多了

**Reply by cheaprYang** [reply]
> 我来吧，搞得差不多了

这个有进展吗？

**Reply by zkwolf** [reply]
功能基本搞完了，文档还得花点时间

**Reply by zkwolf** [reply]
同步一下进度，最近脚骨折了，还有工作有点忙外加有点沉迷原神，进度有点慢2333333（问就是没有任何推进

**Reply by cheaprYang** [reply]
> 同步一下进度，最近脚骨折了，还有工作有点忙外加有点沉迷原神，进度有点慢2333333（问就是没有任何推进

有进展吗

**Reply by kimmy-wang** [reply]
> > 同步一下进度，最近脚骨折了，还有工作有点忙外加有点沉迷原神，进度有点慢2333333（问就是没有任何推进
> 
> 
> 
> 有进展吗

你不看pr吗？今天已经合并了

**Reply by cheaprYang** [reply]
> 

哈哈 还真没看

**Reply by kimmy-wang** [reply]
> > 
> 
> 
> 
> 哈哈 还真没看

😀

**Comment by wsxe9988** [comment]
3.0啥时候能release呢

**Comment by kimmy-wang** [comment]
个人感觉ant-design-vue项目的开发进度相比element-plus等项目比较缓慢

**Comment by wsxe9988** [comment]
https://github.com/ant-design/ant-design/issues/34087

**Comment by lihanspace** [comment]
我在nuxt3中使用`Dropdown`+`Menu`。
组件是可以正常显示的，但是控制台会有警告。

~/layouts/default.vue

```vue
<template>
  ...
    <Dropdown>
      <div>TEST</div>
      <template #overlay>
        <Menu>
          <Menu.Item>TEST1</Menu.Item>
          <Menu.Item>TEST2</Menu.Item>
        </Menu>
      </template>
    </Dropdown>
  ...
</template>
```

控制台警告：

```console
[Vue warn]: Hydration node mismatch:
- Client vnode: Symbol(Comment) 
- Server rendered DOM: "" (text) 
  at <AMenuItem> 
  at <Anonymous> 
  at <AMenu mode="vertical" selectable=false expandIcon=fn<expandIcon2>  ... > 
  at <Align target=fn<bound getRootDomNode> key="popup" ref=Ref< undefined >  ... > 
  at <BaseTransition onBeforeEnter=fn<onBeforeEnter> appear=false persisted=false  ... > 
  at <Transition ref=Ref< undefined > onBeforeEnter=fn<onShowPrepare2> > 
  at <PopupInner prefixCls="ant-dropdown" point=null align= {points: Array(2), overflow: {…}, offset: Array(2), targetOffset: Array(2)}  ... > 
  at <Popup prefixCls="ant-dropdown" destroyPopupOnHide=false visible=true  ... > 
  at <Portal key="portal" getContainer=fn<bound getContainer3> didUpdate=fn<bound handlePortalUpdate> > 
  at <Trigger openClassName=undefined minOverlayWidthMatchTrigger=undefined alignPoint=undefined  ... > 
  at <Anonymous arrow=undefined trigger="hover" visible=undefined  ... > 
  at <ADropdown> 
  at <Header> 
  at <Default > 
  at <AsyncComponentWrapper > 
  at <BaseTransition mode="out-in" appear=false persisted=false  ... > 
  at <Transition name="layout" mode="out-in" > 
  at <NuxtLayout> 
  at <App> 
  at <NuxtRoot>
```

偶尔还会有这个警告：

```console
[Vue warn]: Hydration node mismatch:
- Client vnode: div 
- Server rendered DOM: " TEST " (text) 
  at <Trigger openClassName=undefined minOverlayWidthMatchTrigger=undefined alignPoint=undefined  ... > 
  at <Anonymous arrow=undefined trigger="hover" visible=undefined  ... > 
  at <ADropdown> 
  at <Header> 
  at <Default > 
  at <AsyncComponentWrapper > 
  at <BaseTransition mode="out-in" appear=false persisted=false  ... > 
  at <Transition name="layout" mode="out-in" > 
  at <NuxtLayout> 
  at <App> 
  at <NuxtRoot>
```

**Comment by jingshanchen** [comment]
浏览器器兼容性是否可以考虑chrome60几的版本 ，像element plus chrome版本是 >= 64

**Reply by tangjinzhou** [reply]
try

**Comment by ILfengzi** [comment]
已收到！

**Comment by kimmy-wang** [comment]
ant design pro(vue3)模板是不是该更新了？

**Reply by sendya** [reply]
https://store.antdv.com/pro/preview/user/login

**Reply by kimmy-wang** [reply]
> https://store.antdv.com/pro/preview/user/login

这个支持ant design vue 3.0了吗？还有ssr的问题解决了吗？

**Reply by sendya** [reply]
都支持

**Reply by kimmy-wang** [reply]
> 都支持

之前ssr的情况下，menu菜单不显示，这个问题解决了？

**Reply by tangjinzhou** [reply]
ssr 暂不支持 @sendya

**Reply by tangjinzhou** [reply]
3.0 组件库的 menu 应该没有问题的，pro没有测试

**Reply by kimmy-wang** [reply]
3.0发版了，pro建议增加ssr支持

**Reply by kimmy-wang** [reply]
<img width="1770" alt="截屏2022-04-06 19 33 19" src="https://user-images.githubusercontent.com/19725091/161965659-846ff294-75a3-495c-814e-bb7bb108d9d0.png">

**Reply by wsxe9988** [reply]
后续有什么计划呢，继续对标ant design 更新功能？

**Reply by kimmy-wang** [reply]
@tangjinzhou 我购买pro模板，然后进行ssr适配，如果遇到ssr菜单不显示的问题，我提供最小复现示例，可以帮忙解决吗？

**Comment by ILfengzi** [comment]
已收到！

**Comment by Alone-2** [comment]
V3的DatePicker[picker="quarter"] 支持将季度换成中文吗，目前是[Q1,Q2,Q3,Q4],换为[第一季度,第二季度,第三季度，第四季度]
